scheduler/src/cook/mesos/constraints.clj

```diff
@@ -160,10 +157,13 @@
       (let [vm-resources (.getCurrAvailableResources target-vm)
             vm-attributes (get-vm-lease-attr-map vm-resources)
             [passes? reason] (job-constraint-evaluate constraint vm-resources vm-attributes
-                                                      (into (vec (.getRunningTasks target-vm))
-                                                              (mapv (fn [^com.netflix.fenzo.TaskAssignmentResult result]
-                                                                      (.getRequest result))
-                                                                     (.getTasksCurrentlyAssigned target-vm))))]
+                                                      (loop [tasks-assigned (.getTasksCurrentlyAssigned target-vm)
```

Comment:
Could this be achieved with reduce? Could it benefit from transients (https://clojure.org/reference/transients)? 

Please add a comment why this is a loop so it isn't changed back some point later.